### ISSUE TITLE
Precompiled libs outdated

### ISSUE THREAD

**Issue by RaphaelRoyerRivard** [Blizzard/s2client-api#246]
HI,
I want to use the Connect() method of the Coordinator class but it does not exist in the version of the precompiled libs. Could it be updated please?

Thanks

**Comment by herodrigues** [comment]
A much better option is to add `s2client-api` as a submodule of your project.


Considering you have a `src` folder in your project, you'll need a CMakeLists.txt file like this in your project root folder:
```
cmake_minimum_required(VERSION 3.1)

project(your_binary_name)

# Specify output directories.
set(EXECUTABLE_OUTPUT_PATH ${PROJECT_BINARY_DIR}/bin)

# Build with c++14 support, required by sc2api.
set(CMAKE_CXX_STANDARD 14)

# Allow creating filters for projects in visual studio.
set_property(GLOBAL PROPERTY USE_FOLDERS ON)

# Disable building of examples in the sc2api submodule.
set(BUILD_API_EXAMPLES OFF CACHE INTERNAL "" FORCE)

# Disable building of tests in the sc2api submodule.
set(BUILD_API_TESTS OFF CACHE INTERNAL "" FORCE)

add_subdirectory("s2client-api")
add_subdirectory("src")
```

Now, install Visual Community and cmake. Then, do the following steps:
```
cd <path-to-your-project>
git submodule add https://github.com/Blizzard/s2client-api.git s2client-api
git submodule update --init --recursive
mkdir build 
cd build
cmake ../ -G "Visual Studio 15 2017 Win64"
start your_solution.sln
```

If you're not sure how to do that, check out @alkurbatov's project: https://github.com/alkurbatov/suvorov-bot

**Comment by herodrigues** [comment]
@RaphaelRoyerRivard were you able to do the steps above?

**Comment by Archiatrus** [comment]
The building instructions are in the docs: [docs/building.md](https://github.com/Blizzard/s2client-api/blob/master/docs/building.md).

**Comment by herodrigues** [comment]
@Archiatrus Yeah, I know. I was talking about integrating the API into a personal project that uses s2client-api as dependency. 

This way it's possible to add s2client-api as a git submodule and compile both the API and a personal project (bot). Also, the developer can use the latest API modifications without needing to compile it in a separate way each time a new PR is merged.

**Comment by RaphaelRoyerRivard** [comment]
Yeah, I should do that for my CommandCenter project, but I guess I'll do it only the next time I need new changes.

**Comment by alkurbatov** [comment]
I believe the best solution would be to fully move CommandCenter to cmake (currently it uses cmake for linux+osx and presaved visual studio solution) and s2client-api as a submodule. Since CommandCenter is usually the default ‘starting’ bot that will help much more then any documentation anywhere.
Also we can throw out all these complex instructions regarding installation process :)
